client/App.jsx

```diff
@@ -1,15 +1,127 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import axios from 'axios';
+import CurrentProduct from './CurrentProduct.jsx';
+import FeedList from './FeedList.jsx';
 
+import styled from 'styled-components';
+
+
+///////////styled components///////////////
+// const styledItems = styled.styledItems`
+//   background-color: papayawhip;
+// `;
+
+// grid-template-rows: 50% 1fr 1fr 1fr  1fr;
+
+const Header = styled.div`
+  display: grid;
+  grid-template:
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'img'
+    'Name'
+    'rating'
+    'price'
+    'condition';
+  text-align: center;
+  border: 0.5px solid #f3eeee;
+  font-family: montserratregular,Arial,Helvetica,sans-serif;
+  font-size: 10px;
+  `;
+
+const Name = styled.div`
+  grid-area: Name;
+  border: 0.5px solid #f3eeee;
+  background-color: lightgray;
+`;
+
+const Img = styled.div`
+  grid-area: img;
+  border: 0.5px solid #f3eeee;
+`;
+const Rating = styled.div`
+  grid-area: rating;
+  border: 0.5px solid #f3eeee;
+`;
+const Price = styled.div`
+  grid-area: price;
+  border: 0.5px solid #f3eeee;
+  background-color: lightgray;
+`;
+const Condition = styled.div`
+  grid-area: condition;
+  border: 0.5px solid #f3eeee;
+`;
+
+
+///////////////////////////////////////////
 class App extends React.Component {
-  constructor(props){
-    super(props)
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      similarItems: []
+    };
+
+  }
+
+  //get request receives an object with the property 'data' that has the guitar data
+  componentDidMount() {
+    axios.get('/api/similaritems')
+      .then(data => {
+        var product = [];
+        product.push(data.data);
+        return product;
+      })
+      .then( product => {
+        this.setState({
+          similarItems: product
+        });
+      })
+      .then( () => {
+        this.renderView();
+      });
+  }
+
+
+  renderView() {
+    if (this.state.similarItems.length) {
+      return (
+        <div className='similarItems'>
+          <Header>
+            <Img>  </Img>
+            <Name >Name</Name>
+            <Rating>Rating</Rating>
+            <Price>Price</Price>
+            <Condition>Condition</Condition>
+          </Header>
+          <CurrentProduct product={this.state.similarItems[0][0]} />
+          <FeedList similarItems={this.state.similarItems[0].slice(1, 6)}/>
+        </div>
+      );
+    }
   }
```

Comment:
There's nothing wrong with this approach necessarily, but you may find it helpful to just within your render function deconstruct state (as the airbnb guide expects), and then use a ternary operator to handle your conditional rendering. Either way, you might want to consider returning null or a react fragment in the event your condition isn't met.

Reply:
I was having problems with this too,,,
There would be nothing to render until my GET request (which is in my componentDidMount function) retrieved data, so the initial render attempt would return undefined and throw errors.
I ended up doing this (with no 'if the condition is met' option), so it will only try to render if the GET request completed and there was data to render.

I am assuming that returning null or a react fragment if there is no data will also give me errors..? I will try it out though! 
Otherwise, I need to figure out how to make the render to wait until the GET request is done.. (I don't understand why I've never had this problem in earlier sprints)